src/workers/mempool-watcher.ts

```diff
@@ -0,0 +1,102 @@
+/**
+ * AR.IO Gateway
+ * Copyright (C) 2022-2023 Permanent Data Solutions, Inc. All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import { default as wait } from 'wait';
+import * as winston from 'winston';
+import { ArweaveCompositeClient } from '../arweave/composite-client.js';
+import { TransactionFetcher } from './transaction-fetcher.js';
+
+export class MempoolWatcher {
+  // Dependencies
+  private log: winston.Logger;
+  private arweaveClient: ArweaveCompositeClient;
+  private txFetcher: TransactionFetcher;
+
+  // Parameters
+  private mempoolPoolingIntervalMs: number;
+
+  // State
+  private shouldRun: boolean;
+  private pendingTxs: Map<string, boolean> = new Map();
+
+  constructor({
+    log,
+    arweaveClient,
+    txFetcher,
+    mempoolPoolingIntervalMs,
+  }: {
+    log: winston.Logger;
+    arweaveClient: ArweaveCompositeClient;
+    txFetcher: TransactionFetcher;
+    mempoolPoolingIntervalMs: number;
+  }) {
+    this.log = log.child({ class: 'MempoolWatcher' });
+    this.arweaveClient = arweaveClient;
+    this.txFetcher = txFetcher;
+    this.mempoolPoolingIntervalMs = mempoolPoolingIntervalMs;
+    this.shouldRun = false;
+  }
+
+  private normalizePendingTxs(txs: string[]): void {
+    const mempoolTxs = new Set(txs);
```

Comment:
I transformed it to a `Set` because it's faster to iterate as we don't know the size of this list, but maybe this is not required

Reply:
I'm fine either way on this one. I suspect an array actually ends up being faster, but I don't think we should worry about it too much.

Reply:
My concern with this `Set` is memory as we keep the array and the Set for a small period of time. So maybe just iterating over the array would be better considering memory usage